src/app/common/modals/tabswitcher.tsx

```diff
@@ -189,21 +215,32 @@ class TabSwitcherModal extends React.Component<{}, {}> {
     @mobx.computed
     @boundMethod
     filterOptions(searchInput: string): SwitcherDataType[] {
+        const searchLower = searchInput.toLowerCase();
         let filteredScreens = [];
 
-        for (let i = 0; i < this.options.length; i++) {
-            let tab = this.options[i];
+        // Combine both arrays for iteration
+        const combinedOptions = [...this.options, ...additionalOptions];
```

Comment:
I feel like we should just iterate through these arrays separately, since the logic inside the for loop is different for `this.options` and `additionalOptions`

Reply:
addressed